router/apiHandler.go

```diff
@@ -96,14 +96,20 @@ func ApiViewHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func ApiUploadHandler(w http.ResponseWriter, r *http.Request) {
-	if config.UploadsDisabled {
-		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
-		return
-	}
-
 	token := r.Header.Get("Authorization")
 	user := model.User{}
 	db.ORM.Where("api_token = ?", token).First(&user) //i don't like this
+	if config.UploadsDisabled && config.AdminsAreStillAllowedTo && user.Status != 2 && config.TrustedUsersAreStillAllowedTo && user.Status != 1 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	} else if config.UploadsDisabled && !config.AdminsAreStillAllowedTo && user.Status == 2 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	} else if config.UploadsDisabled && !config.TrustedUsersAreStillAllowedTo && user.Status == 1 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	}
```

Comment:
Hm, the config.UploadsDisabled should probably be moved to a seperate if-clause for better readability